beacon-chain/db/deposits.go

```diff
@@ -82,6 +83,28 @@ func (db *BeaconDB) AllDeposits(ctx context.Context, beforeBlk *big.Int) []*pb.D
 	return deposits
 }
 
+// DepositsContainersTillBlock returns the deposit containers sorted by block height where
+// block height lower then beforeBlk.
+func (db *BeaconDB) DepositsContainersTillBlock(ctx context.Context, beforeBlk *big.Int) []*DepositContainer {
+	ctx, span := trace.StartSpan(ctx, "BeaconDB.AllDepositsWithTrieRoots")
+	defer span.End()
+	db.depositsLock.RLock()
+	defer db.depositsLock.RUnlock()
+
+	var deposits []*DepositContainer
+	for _, ctnr := range db.deposits {
+		if beforeBlk == nil || beforeBlk.Cmp(ctnr.block) > -1 {
+			deposits = append(deposits, ctnr)
+		}
+	}
+	// Sort the deposits by Merkle index in an ascending order.
+	sort.SliceStable(deposits, func(i, j int) bool {
```

Comment:
If we keep the deposits sorted on insert, this read operation would be much faster.

Reply:
as i changed the behavior into one time read and sort per call. i prefer not to depend upon insertion order

Reply:
Why not? I don't understand the redundant sorting on reads. 
Can we cache it at least?

Reply:
sorted on insertion